languages/csharp/exercises/concept/user-defined-exceptions/.meta/Example.cs

```diff
@@ -0,0 +1,68 @@
+using System;
+
+public class CalculationException : Exception
+{
+    public CalculationException(int operand1, int operand2, string message, Exception inner) : base(message, inner)
+    // TODO: complete the definition of the constructor
+    {
+        Operand1 = operand1;
+        Operand2 = operand2;
+    }
+
+    public int Operand1 { get; }
+    public int Operand2 { get; }
+}
+
+public class CalculatorTestHarness
+{
+    private Calculator calculator;
+
+    public CalculatorTestHarness(Calculator calculator)
+    {
+        this.calculator = calculator;
+    }
+
+    public string Multiply(int x, int y)
+    {
+        try
+        {
+            TestMultiplication(x, y);
+        }
+        catch (CalculationException cex) when (cex.Operand1 < 0)
+        {
+            return "Multiply failed for a negative value. " + cex.InnerException.Message;
```

Comment:
Maybe we can also include the operands in the returned message?

Reply:
A) I agree that one would expect to see the operand returned.  My rationale was that including the operands would add complexity and increase the number of representations.  The story had already an artificial turn by only testing a single operand and caring about negativity in the first place.

Reply:
> The story had already an artificial turn by only testing a single operand and caring about negativity in the first place.

Good point. I was trying to come up with a way to also use `Operand2` somewhere. Maybe do `(cex.Operand1 < 0 || cex.Operand2 < 0)`? I don't think that will hugely affect the number of representations, do you?

Reply:
Something similar done.